src/kixi/stats/distribution.cljc

```diff
@@ -604,38 +604,45 @@
 
 (defn uniform
   "Returns a uniform distribution.
-  Params: {:a ∈ ℝ, :b ∈ ℝ}"
+  Params: {:a ∈ ℝ, :b ∈ ℝ, :a < :b}"
   [{:keys [a b]}]
+  (assert (< a b) (str "a (" a ") must be less than b (" b ")."))
   (->Uniform a b))
 
 (defn exponential
   "Returns an exponential distribution.
   Params: {:rate ∈ ℝ > 0}"
   [{:keys [rate]}]
+  (assert (pos? rate) (str "rate (" rate ") must be positive."))
   (->Exponential rate))
 
 (defn bernoulli
   "Returns a Bernoulli distribution.
   Params: {:p ∈ [0 1]}"
   [{:keys [p]}]
+  (assert (<= 0.0 p 1.0) (str "p (" p ") must be between 0.0 and 1.0."))
   (->Bernoulli p))
 
 (defn binomial
   "Return a binomial distribution.
   Params: {:n ∈ ℕ, :p ∈ [0 1]}"
   [{:keys [n p]}]
+  (assert (nat-int? n) (str "n (" n ") must be a natural number."))
+  (assert (<= 0.0 p 1.0) (str "p (" p ") must be between 0.0 and 1.0."))
   (->Binomial n p))
 
 (defn normal
   "Returns a normal distribution.
-  Params: {:location ∈ ℝ, :scale ∈ ℝ}"
+  Params: {:location ∈ ℝ, :scale ∈ ℝ > 0}"
   [{:keys [location scale mu sd]}]
+  (assert (pos? (or scale sd)) (str "scale/sd (" (or scale sd) ") must be positive."))
   (->Normal (or location mu) (or scale sd)))
 
 (defn t
   "Returns a t distribution.
-  Params: {:v ∈ ℕ > 0}"
+  Params: {:v ∈ ℝ > 0}"
   [{:keys [v]}]
+  (assert (pos? v) (str "v (" v ") must be positive."))
   (->T v))
 
 (defn gamma
```

Comment:
OOI, why skip the gamma and beta distributions?

Reply:
I was going by the docstring, which did not specify any constraints.
I've now updated the docstrings and added the asserts.